runtime/io/common/src/aws/smithy/kotlin/runtime/io/internal/BufferOperations.kt

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+@file:Suppress("NOTHING_TO_INLINE")
+
+package aws.smithy.kotlin.runtime.io.internal
+
+import aws.smithy.kotlin.runtime.io.*
+
+internal inline fun SdkBuffer.commonSkip(byteCount: Long) = inner.skip(byteCount)
+
+internal inline fun SdkBuffer.commonReadByte(): Byte = inner.readByte()
+
+internal inline fun SdkBuffer.commonReadShort(): Short = inner.readShort()
+
+internal inline fun SdkBuffer.commonReadShortLe(): Short = inner.readShortLe()
+
+internal inline fun SdkBuffer.commonReadLong(): Long = inner.readLong()
+
+internal inline fun SdkBuffer.commonReadLongLe(): Long = inner.readLongLe()
+
```

Comment:
It's because `SdkBuffer`, `SdkBufferedSource`, and `SdkBufferedSink` are `expect`/`actual` and `expect` can't have default implementations. You'll see this pattern in both `okio` and `ktor-io` (and probably other libraries).